src/app/singletons/item_manager.rs

```diff
@@ -0,0 +1,98 @@
+use crate::*;
+
+mod impl_ {
+    use crate::*;
+
+    extern "C" {
+        #[link_name = "_ZN3lib9SingletonIN3app12StageManagerEE9instance_E"]
+        pub(super) static INSTANCE: *mut app::ItemManager;
+
+        #[link_name = "_ZN3app8lua_bind47ItemManager__find_active_item_from_area_id_implEPNS_11ItemManagerEj"]
+        pub(super) fn find_active_item_from_area_id(manager: *const app::ItemManager, arg2: u32) -> *const skyline::libc::c_void;
+
+        #[link_name = "_ZN3app8lua_bind42ItemManager__find_active_item_from_id_implEPNS_11ItemManagerEj"]
+        pub(super) fn find_active_item_from_id(manager: *const app::ItemManager, battle_object_id: u32) -> *const skyline::libc::c_void;
+
+        #[link_name = "_ZN3app8lua_bind33ItemManager__get_active_item_implEPNS_11ItemManagerEm"]
+        pub(super) fn get_active_item(manager: *const app::ItemManager, arg2: u64) -> *const skyline::libc::c_void;
+
+        #[link_name = "_ZN3app8lua_bind44ItemManager__get_num_of_active_item_all_implEPNS_11ItemManagerE"]
+        pub(super) fn get_num_of_active_item_all(manager: *const app::ItemManager) -> usize;
+
+        #[link_name = "_ZN3app8lua_bind41ItemManager__get_num_of_ownered_item_implEPNS_11ItemManagerEjNS_8ItemKindE"]
+        pub(super) fn get_num_of_ownered_item(manager: *const app::ItemManager, battle_object_id: u32, item_kind: app::ItemKind) -> usize;
+
+        #[link_name = "_ZN3app8lua_bind52ItemManager__is_change_fighter_restart_position_implEPNS_11ItemManagerE"]
+        pub(super) fn is_change_fighter_restart_position(manager: *const app::ItemManager) -> bool;
+
+        #[link_name = "_ZN3app8lua_bind37ItemManager__remove_item_from_id_implEPNS_11ItemManagerEj"]
+        pub(super) fn remove_item_from_id(manager: *const app::ItemManager, battle_object_id: u32);
+    }
+}
+
+#[repr(C)]
+pub struct ItemManager {}
+
+impl ItemManager {
+    pub fn instance() -> Option<&'static Self> {
```

Comment:
Maybe change this to `as_ref()` as you did on the FighterManager calls? If you wouldn't mind doing that in the other places where I write this boilerplate, I would appreciate it. I didn't know that call was an option and that is a much nicer way of handling this.

Reply:
Sure thing, I'll take care of that in my next commit. I can't believe I didn't think about that.